### ISSUE TITLE
Enable TestContext output in Portable

### ISSUE THREAD

**Issue by CharliePoole** [nunit/nunit#2148]
As it now stands, any output lines in a test require conditional compilation, like this...

```C#
#if !PORTABLE
    Console.WriteLine("Some message");
    TestContext.Progress.WriteLine("Progress message");
#endif
```

This is kind of annoying. We can't do anything about Console, but we could define the output properties in portable in such a way that they compile but do nothing.

I realize we want Portable to go away, but that makes it an even better idea IMO, since it eliminates some uses of `#if !PORTABLE` immediately.

**Comment by jnm2** [comment]
What about something like this? It has the benefit that the compiler actually removes the entire call site using `Utils.WriteMessage` instead of making a call to a dummy method.

```c#
    internal static class Utils
    {
        [Conditional("!PORTABLE")]
        public static void WriteMessage(string message)
        {
            Console.WriteLine(message);
            TestContext.Progress.WriteLine(message);
        }
    }
```

**Comment by rprouse** [comment]
I have started converting the portable build to .NET Standard. Once that is done, we should be able to re-enable The TestContext calls at least.

@jnm2 great idea about the attribute. I'd forgotten that was available.

**Comment by CharliePoole** [comment]
@rprouse So close this as Won't Fix?

**Comment by jnm2** [comment]
What's the status on this?

**Comment by CharliePoole** [comment]
I'd close it, since we no longer have a Portable build. The alternative is to convert it to be about the .NET Standard builds, but it has a load of discussion about portable, which might be confusing. IMO a new issue would make sense.
